src/System.IO.FileSystem.Watcher/ref/System.IO.FileSystem.Watcher.cs

```diff
@@ -46,6 +46,12 @@ public partial class FileSystemWatcher
         public System.IO.WaitForChangedResult WaitForChanged(System.IO.WatcherChangeTypes changeType) { return default(System.IO.WaitForChangedResult); }
         public System.IO.WaitForChangedResult WaitForChanged(System.IO.WatcherChangeTypes changeType, int timeout) { return default(System.IO.WaitForChangedResult); }
     }
+    public partial class InternalBufferOverflowException : System.Exception
+    {
+        public InternalBufferOverflowException() { }
+        public InternalBufferOverflowException(string message) { }
+        public InternalBufferOverflowException(string message, System.Exception inner) { }
+    }
```

Comment:
Since you're adding this, can you add it with the appropriate serialization as well? It should have a protected ctor:
```C#
protected InternalBufferOverflowException(SerializationInfo info, StreamingContext context) : base (info, context) { }
```
Also, the base type should be SystemException, right?